NoteFind/View Controllers/SearchViewController.m

```diff
@@ -5,104 +5,106 @@
+    loadingMoreView = [[InfiniteScrollActivityView alloc] initWithFrame:frame];
+    loadingMoreView.hidden = true;
+    [self.tableView addSubview:loadingMoreView];
     
+    UIEdgeInsets insets = self.tableView.contentInset;
+    insets.bottom += InfiniteScrollActivityView.defaultHeight;
+    self.tableView.contentInset = insets;
 }
 
-- (void) getTags {
-    PFQuery *tagQuery = [Tags query];
-    [tagQuery orderByAscending:@"title"];
-    [tagQuery includeKey:@"title"];
-    NSMutableArray *tagTitles = [[NSMutableArray alloc]init];
-    
-    [tagQuery findObjectsInBackgroundWithBlock:^(NSArray<Tags *> *tags, NSError *error) {
-        if (tags) {
-            // adding title to allTags for table view
-            for (Tags *tag in tags) {
-                NSString *title = tag.title;
-                [tagTitles addObject:title];
-
-                [self.allTags addObject:tag];
-            }
-            self.allTagTitles = (NSArray *)tagTitles;
-
-           
-        } else {
-            // TODO: handle error
-        }
-        [self.tableView reloadData];
-    }];
-}
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
-        
-    if (self.isFiltered) {
-
-        cell.textLabel.text = self.displayTagTitles[indexPath.row];
-    }
-    
-    else {
-        cell.textLabel.text = self.allTagTitles[indexPath.row];
-
-
-    }
-    
+    Tags *tag = self.displayTags[indexPath.row];
+    cell.textLabel.text = tag.title;
     return cell;
     
 }
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-    if (self.isFiltered) {
-
-        return self.displayTagTitles.count;
-    }
-    
-    return self.allTagTitles.count;
-
+    return self.displayTags.count;
 }
 
 - (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
-    if (searchText.length == 0) {
-        self.isFiltered = NO;
-    } else {
-        self.isFiltered = YES;
+    //removing white space
+    self.textToSearch = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+    self.setsLoaded = 0;
+    [self loadTags:self.textToSearch];
+}
 
-        self.displayTagTitles = [[NSMutableArray alloc]init];
-        for (NSString *tag in self.allTagTitles) {
-            NSRange range = [tag rangeOfString:searchText options:NSCaseInsensitiveSearch];
-            if (range.location != NSNotFound) {
-                [self.displayTagTitles addObject:tag];
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+    if (!isMoreDataLoading) {
+        int scrollViewContentHeight = self.tableView.contentSize.height;
+        int scrollOffsetThreshold = scrollViewContentHeight - self.tableView.bounds.size.height;
+
+        if (scrollView.contentOffset.y > scrollOffsetThreshold && self.tableView.isDragging) {
```

Comment:
do we need to check if we're already loading data here?

Reply:
i did this because when i get more data, the array is replaced with the new data that's queried. so if i don't check it might be overriding the data that's in the array already